http4s-client-monix-catnap/src/main/scala/com/avast/sst/http4s/client/monix/catnap/Http4sClientCircuitBreakerModule.scala

```diff
@@ -0,0 +1,45 @@
+package com.avast.sst.http4s.client.monix.catnap
+
+import cats.effect.{Clock, Resource, Sync}
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State.{Closed, HalfOpen, Open}
+import com.avast.sst.monix.catnap.{CircuitBreakerConfig, CircuitBreakerMetrics, CircuitBreakerModule}
+import org.http4s.Response
+import org.http4s.client.Client
+import org.slf4j.LoggerFactory
+
+object Http4sClientCircuitBreakerModule {
+
+  private lazy val logger = LoggerFactory.getLogger("Http4sClientCircuitBreakerModule")
+
+  /**  */
+  def make[F[_]: Sync](name: String,
+                       circuitBreakerConfig: CircuitBreakerConfig,
```

Comment:
it would probably also be a good idea to pass an actual `CircuitBreaker[F]` instance here (instead of passing config and creating it here on your own)

Reply:
This module is supposed to give you "something more". In this case it's logging and metrics for the circuit breaker therefore I need to instantiate it here. I think there are 2 kinds of modules - base modules and these "interop" modules which give you something on top of just the base modules.

Reply:
I see.
What do you think about splitting creation of the circuit breaker, and wrapping HTTP client with that circuit breaker?

Reply:
for the record, I in no way object to introducing a new helper method, which would combine these 2 smaller ones

It's just that I think that it's better to have small, versatile building blocks, and build bigger things from them, compared to having just big unwieldy bricks.